site/zenodo_rdm/redirector.py

```diff
@@ -0,0 +1,221 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2023 CERN.
+#
+# ZenodoRDM is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""Redirector functions and rules."""
+
+from flask import request, url_for
+
+ZENODO_TYPE_SUBTYPE_LEGACY = {
+    "publications": "publication",
+    "books": "publication.book",
+    "books-sections": "publication.section",
+    "conference-papers": "publication.conferencepaper",
+    "journal-articles": "publication.article",
+    "patents": "publication.patent",
+    "preprints": "publication.preprint",
+    "deliverable": "publication.deliverable",
+    "milestone": "publication.milestone",
+    "proposal": "publication.proposal",
+    "reports": "publication.report",
+    "theses": "publication.thesis",
+    "technical-notes": "publication.technicalnote",
+    "working-papers": "publication.workingpaper",
+    "other-publications": "publication.other",
+    "posters": "poster",
+    "presentations": "presentation",
+    "datasets": "dataset",
+    "images": "image",
+    "figures": "image.figure",
+    "drawings": "image.drawing",
+    "diagrams": "image.diagram",
+    "photos": "image.photo",
+    "other-images": "image.other",
+    "videos": "video",
+    "software": "software",
+    "lessons": "lesson",
+    "physicalobject": "physicalobject",
+    "workflows": "workflow",
+    "other": "other",
+}
+
+
+def communities_detail_view_function():
+    """Implements redirector view function for communities detail.
+
+    The following routes are redirected as follows:
+        - /communities/about/<id>/ -> GET /communities/<pid_value>
+        - /collection/user-<id> -> GET /communities/<pid_value>
+        - /communities/<community_id>/about -> GET /communities/<pid_value>
+
+    :return: url for the view 'invenio_app_rdm_communities.communities_detail'
+    :rtype: str
+    """
+    _id = request.view_args.get("id", request.view_args.get("community_id"))
```

Comment:
Don't you want to make the `community_id` mandatory? That `.get` can be `None`?
```suggestion
    _id = request.view_args.get("id", request.view_args["community_id"])
```
Comment valid for all the methods below.

Reply:
I had only moved code, not evaluated its correctness. Thanks! I've modified the access :)